app/src/main/java/hr/foi/air/sportloc/view/ui/fragment/CommentListFragment.java

```diff
@@ -0,0 +1,191 @@
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+
+
+import hr.foi.air.sportloc.service.model.ActiveUserModel;
+import hr.foi.air.sportloc.service.model.CommentModel;
+import hr.foi.air.sportloc.service.model.ModelEnum;
+import hr.foi.air.sportloc.service.model.UserModel;
+import hr.foi.air.sportloc.view.ui.activity.ProfileActivity;
+import hr.foi.air.sportloc.view.util.IntentManager;
+import hr.foi.air.sportloc.viewmodel.CommentsViewModel;
+import hr.foi.air.sportloc.databinding.FragmentCommentListBinding;
+
+
+public class CommentListFragment extends Fragment {
+    private Unbinder unbinder;
+    private CommentsListViewListener listener;
+    private String stringId;
+
+    @BindView(R.id.lv_comment)
+    ListView listView;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        FragmentCommentListBinding viewBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_comment_list, container, false);
+        View view = viewBinding.getRoot();
+        unbinder = ButterKnife.bind(this, view);
+
+        Map<String, Object> map = new HashMap<>();
+        map.put(ModelEnum.UserModel.name(), ActiveUserModel.getInstance().getActiveUser());
+
+
+        Bundle bundle = this.getArguments();
+        stringId=bundle.getString("stringId");
+        int id=Integer.parseInt(stringId);
+        showComments(id);
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+    }
+
+
+    public interface CommentsListViewListener {
+        void onOpenCommentsList();
+    }
+
+    public void setListener(CommentsListViewListener listener) {
+        this.listener = listener;
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        unbinder.unbind();
+    }
+
+    public void showComments(Integer id) {
+        CommentsViewModel commentsViewModel = new CommentsViewModel();
+        commentsViewModel.getComments(id);
+
+        commentsViewModel.getCommentsObservable().observe(getActivity(), result -> {
+            int commentCounter = 0;
+            List<CommentModel> list = result;
+
+            for (int i = 0; i < list.size(); i++) {
+                commentCounter++;
+            }
+
+            String usernameListArray[] = new String[commentCounter];
+
+            for (int i = 0; i < list.size(); i++) {
+                CommentModel comment;
+                comment = list.get(i);
+                int commentatorId = comment.getCommentatorId();
+                boolean vote = comment.isVote();
+                String userComment = comment.getComment();
+                String commentatorUsername = comment.getCommentator();
+
+                usernameListArray[i] = commentatorUsername;
+            }
```

Comment:
this seems useless... You do not even have to use an array, you can just use a lambda expression such as 
list.stream().map(x -> x.getCommentator).collect(Collectors.toList());

Reply:
this can still be done in one line.